src/config.ts

```diff
@@ -0,0 +1,109 @@
+import { relative } from 'pathe'
+import { addTemplate, createResolver, useNuxt } from '@nuxt/kit'
+import loadConfig from 'tailwindcss/loadConfig.js'
+import logger from './logger'
+import { resolveModulePaths } from './resolvers'
+import type { ModuleHooks, ModuleOptions, TWConfig } from './types'
+import configMerger from './runtime/merger.mjs'
+
+type PartialTWConfig = Partial<TWConfig>
+
+const makeHandler = <T extends PartialTWConfig>(proxyConfig: T, path: string[] = []): ProxyHandler<T> => ({
+  get: (target, key: string) => {
+    return (typeof target[key] === 'object' && target[key] !== null)
+      ? new Proxy(target[key], makeHandler(proxyConfig, path.concat(key)))
+      : target[key]
+  },
+
+  set(target, key: string, value) {
+    if ((Array.isArray(target) && key === 'length') || JSON.stringify(target[key]) === JSON.stringify(value)) {
+      return Reflect.set(target, key, value)
+    }
+
+    let result = typeof target[key] === 'object' && !Array.isArray(target[key])? () => value : value;
+    (Array.isArray(target) ? path : path.concat(key)).reverse().forEach((k, idx) => {
+      result = { [k]: Array.isArray(target) && idx === 0 ? [result] : result }
+    })
+
+    // @ts-ignore
+    Object.entries(configMerger(proxyConfig, result)).forEach(([k, v]) => proxyConfig[k] = v)
+    return Reflect.set(target, key, value)
+  }
+})
+
+const serializeConfig = <T extends PartialTWConfig>(config: T) =>
+  JSON.stringify(
+    configMerger({ plugins: () => [] }, config),
+    (_, v) => typeof v === 'function' ? `() => (${JSON.stringify(v())})` : v).replace(/"(\(\) => \(.*\))"/g, (_, substr) => substr.replace(/\\"/g, '"')
+  )
+
+/**
+ * Loads all possible Tailwind CSS configuration for a Nuxt project.
+ *
+ * @param moduleOptions configuration for the module
+ * @param nuxt nuxt app
+ * 
+ * @returns template with all configs handled
+ */
+export default async function loadTwConfig(moduleOptions: ModuleOptions, nuxt = useNuxt()) {
+  const [configPaths, contentPaths] = await resolveModulePaths(moduleOptions.configPath, nuxt)
+  const configProxies: [string, PartialTWConfig][] = configPaths.map((configPath) => [configPath, {}])
+
+  const tailwindConfig = await Promise.all((
+    configPaths.map(async (configPath, idx, paths) => {
+      let _tailwindConfig: PartialTWConfig | undefined
+
+      try {
+        _tailwindConfig = loadConfig(configPath)
+      } catch (e) {
+        logger.warn(`Failed to load Tailwind config at: \`./${relative(nuxt.options.rootDir, configPath)}\``, e)
+      }
+
+      // Transform purge option from Array to object with { content }
+      if (_tailwindConfig && !_tailwindConfig.content) {
+        configProxies[idx][1].content = _tailwindConfig.purge
+      }
+
+      await nuxt.callHook('tailwindcss:loadConfig', _tailwindConfig && new Proxy(_tailwindConfig, makeHandler(configProxies[idx][1])), configPath, idx, paths)
```

Comment:
constantly had thoughts to drop this hook, but it also makes sense to keep here as we need to load these configs for the main `tailwindcss:config` hook, otherwise we'd not do any loading at all and have incredible performance.